account/api/serializers.py

```diff
@@ -1,8 +1,100 @@
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
 from rest_framework import serializers
-from account.models import PhonePrefix, CustomUser
+
+
+from account.models import CustomUser, PhonePrefix, Warehouse
+from account.tokens import account_activation_token
+
 
 class PhonePrefixSerializer(serializers.ModelSerializer):
-     
-     class Meta:
+    class Meta:
         model = PhonePrefix
-        fields = ['prefix_number']
+        fields = "__all__"
+
+    def create(self, validated_data):
+        return PhonePrefix.objects.create(**validated_data)
+
+
+class BranchSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Warehouse
+        fields = "__all__"
+
+    def create(self, validated_data):
+        return Warehouse.objects.create(**validated_data)
+
+
+class RegistrationSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(
+        write_only=True, required=True, style={"input_type": "password"}
+    )
+    phone_prefix = PhonePrefixSerializer(many=False)
+    branch = BranchSerializer(many=False)
+
+    class Meta:
+        model = CustomUser
+        fields = (
+            "email",
+            "password",
+            "first_name",
+            "last_name",
+            "gender",
+            "phone_prefix",
+            "phone",
+            "gov_id_prefix",
+            "gov_id",
+            "pin_code",
+            "birth_date",
+            "branch",
+        )
+
+    def validate(self, attrs):
+        password = attrs.get("password")
+        password_confirm = self.context.get("request").data.get("password_confirm")
```

Comment:
Validate gender, phone, phone_prefix, name and surname. It should give API error, not db error that you wrote already

Reply:
Why do we need validation for certain fields, ex (first and last name). why do we need API error?

Reply:
Because, frontend should catch those errors and show them to the user. So errors should be both translatable and user friendly